esiclient/v1/node_network.py

```diff
@@ -79,7 +79,8 @@ def take_action(self, parsed_args):
                 neutron_port = next((np for np in neutron_ports
                                      if np.id == neutron_port_id), None)
 
-                network_id = neutron_port.network_id
+                if neutron_port:
+                    network_id = neutron_port.network_id
```

Comment:
This block of code inside the "if neutron_port_id" in line 78 is essentially saying "if the network exists, find network information"; otherwise it goes to line 102/103 and puts in a bunch of 'None' instead of network information.

What's happening is that we've discovered a new case when the network secretly doesn't exist - when the assumed neutron_port also doesn't exist. The 'if' statement you've added checks that, but it doesn't redirect the code into the 'no network' case. That's why eventually it fails at line 90; that line still assumes a network exists... and it doesn't.

Reply:
Okay... I have a much clearer picture now...
I tried to redirect the code to no network case from the if statement that I added...however, it's resulting into more errors and failure of unit tests....any hint how I can do that?